pymc/model.py

```diff
@@ -1441,6 +1441,9 @@ def add_random_variable(self, var, dims: Optional[Tuple[Union[str, None], ...]]
             if isinstance(dims, str):
                 dims = (dims,)
             assert all(dim in self.coords or dim is None for dim in dims)
+            assert all(
+                var.name != dim for dim in dims
+            ), f"Variable `{var.name}` has the same name as its dimension label."
```

Comment:
Since asserts can be turned off globally, and because of coverage calculation, the lines above should be an if statement raising an exception such as ValueError.
I'm sorry to nitpick about this, since clearly you were just following the style of the line above.. Could you change both into an if/raise?